grpc/src/main/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandlerBuilder.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.grpc;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+
+import org.curioswitch.common.protobuf.json.MessageMarshaller;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.protobuf.Message;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+/**
+ * Constructs a {@link UnframedGrpcErrorHandler} to handle unframed gRPC errors.
+ */
+public final class UnframedGrpcErrorHandlerBuilder {
+    private UnframedGrpcStatusMappingFunction statusMappingFunction = UnframedGrpcStatusMappingFunction.of();
+
+    @Nullable
+    private MessageMarshaller jsonMarshaller;
+
+    private final List<Message> messages = new ArrayList<>();
+
+    private final List<Class<? extends Message>> messageTypes = new ArrayList<>();
+
+    @Nullable
+    private Set<UnframedGrpcErrorResponseType> responseTypes;
+
+    UnframedGrpcErrorHandlerBuilder() {}
+
+    /**
+     * Sets a custom JSON marshaller to be used by the error handler.
+     *
+     * <p>This method allows the caller to specify a custom JSON marshaller
+     * for encoding the error responses. If messages or message types have
+     * already been registered, calling this method will result in an
+     * {@link IllegalArgumentException}. If nothing is specified,
+     * {@link UnframedGrpcErrorHandlers#ERROR_DETAILS_MARSHALLER} is used as
+     * default json marshaller.
+     *
+     * @param jsonMarshaller The custom JSON marshaller to use
+     */
+    public UnframedGrpcErrorHandlerBuilder jsonMarshaller(MessageMarshaller jsonMarshaller) {
+        requireNonNull(jsonMarshaller, "jsonMarshaller");
+        if (!messages.isEmpty() || !messageTypes.isEmpty()) {
+            throw new IllegalArgumentException(
+                    "Cannot set a custom JSON marshaller because one or more Message or MessageType instances" +
+                    " have already been registered. To set a custom marshaller, ensure that no Message or" +
+                    " MessageType registrations have been made before calling this method.");
+        }
+        this.jsonMarshaller = requireNonNull(jsonMarshaller, "jsonMarshaller");
```

Comment:
No need to check nullness twice
```suggestion
        this.jsonMarshaller = jsonMarshaller;
```